ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractArcticOptimizePlan.java

```diff
@@ -208,38 +212,37 @@ protected void addChangeFilesIntoFileTree() {
     LOG.debug("{} start {} plan change files", tableId(), getOptimizeType());
     ChangeTable changeTable = arcticTable.asKeyedTable().changeTable();
 
-    Set<String> changeFiles = new HashSet<>();
-    List<PrimaryKeyedFile> unOptimizedChangeFiles = changeFileScanTasks.stream().map(task -> {
-      PrimaryKeyedFile file = task.file();
+    Set<String> changeFileSet = new HashSet<>();
+    List<ContentFileWithSequence<?>> unOptimizedChangeFiles = this.changeFiles.stream().map(file -> {
       String partition = changeTable.spec().partitionToPath(file.partition());
       currentPartitions.add(partition);
-      if (changeFiles.contains(file.path().toString())) {
+      if (changeFileSet.contains(file.path().toString())) {
         return null;
       }
-      changeFiles.add(file.path().toString());
+      changeFileSet.add(file.path().toString());
       return file;
     }).filter(Objects::nonNull).collect(Collectors.toList());
 
-    long maxTransactionIdLimit = getMaxTransactionIdLimit(unOptimizedChangeFiles);
+    long maxSequenceLimit = getMaxSequenceLimit(unOptimizedChangeFiles);
 
     AtomicInteger addCnt = new AtomicInteger();
     unOptimizedChangeFiles.forEach(file -> {
-      long transactionId = file.transactionId();
+      long sequenceNumber = file.getSequenceNumber();
 
-      if (transactionId >= maxTransactionIdLimit) {
+      if (sequenceNumber >= maxSequenceLimit) {
         return;
       }
       String partition = changeTable.spec().partitionToPath(file.partition());
-      putChangeFileIntoFileTree(partition, file, file.type(), transactionId);
-      markChangeTransactionId(partition, transactionId);
+      putChangeFileIntoFileTree(partition, file);
+      markChangeSequence(partition, file.getSequenceNumber());
 
       addCnt.getAndIncrement();
     });
     LOG.debug("{} ==== {} add {} change files into tree, total files: {}." + " After added, partition cnt of tree: {}",
         tableId(), getOptimizeType(), addCnt, unOptimizedChangeFiles.size(), partitionFileTree.size());
   }
 
-  private long getMaxTransactionIdLimit(List<PrimaryKeyedFile> unOptimizedChangeFiles) {
+  private long getMaxSequenceLimit(List<ContentFileWithSequence<?>> unOptimizedChangeFiles) {
```

Comment:
Minor optimize change files with the order of the change file sequence will result in changes files with larger transaction Id are merged before changes files with smaller transaction Id when streaming and batch writing occur simultaneously.

How should we solve this problem?